service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -333,11 +295,29 @@ public ResponseEntity<AbortRunSetResponse> abortRunSet(UUID runSetId) {
       throw new ForbiddenException(SamService.WRITE_ACTION, SamService.RESOURCE_TYPE_WORKSPACE);
     }
 
+    // Get the run set associated with runSetId
+    RunSet runSet = runSetDao.getRunSet(runSetId);
+
+    // If RunSet is in Queued state, it means that CBAS is still processing the POST request in the
+    // async method. In this case, if the RunSet is set to Cancelling, it might leave the RunSet in
+    // weird state as it's possible that the background thread could still be processing Runs
+    // when the abort request came in and as a result some Runs that had been submitted to
+    // Cromwell got cancelled but some Runs that were still in process of being submitted do in
+    // fact get submitted to Cromwell and are started by Cromwell. And hence the RunSet might have
+    // Runs both in Aborted and Running state. Instead, don't abort RunSet when in Queued state.
+    if (runSet.status() == CbasRunSetStatus.QUEUED) {
+      String errorMessage =
+          "Run Set can't be aborted when it is Queued state as system might still be processing the request.";
```

Comment:
I'd leave out the why. I think just "Run Set can't be aborted when it is Queued state" is fine